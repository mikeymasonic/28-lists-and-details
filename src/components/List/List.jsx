import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ name, occupation, status, image }) => (
  <ul key={name}>
    <li><Link to={`detail/${name}`} >{name}</Link></li>
    <li>{occupation}</li>
    <li>{status}</li>
    <img src={image} />
  </ul>
);

List.propTypes = {
  name: PropTypes.string.isRequired,
  occupation: PropTypes.string,
  status: PropTypes.string,
  image: PropTypes.string,

};

export default List;
