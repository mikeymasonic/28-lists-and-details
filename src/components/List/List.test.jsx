import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List
      name="Fox Mulder"
      occupation="FBI Agent"
      status="Alive"
      image="https://vignette.wikia.nocookie.net/x-files/images/c/c0/TXF_Mulder.jpg/revision/latest/scale-to-width-down/310?cb=20171214040226"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
