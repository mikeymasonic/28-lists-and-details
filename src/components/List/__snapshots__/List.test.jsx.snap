// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul
  key="Fox Mulder"
>
  <li>
    <Link
      to="detail/Fox Mulder"
    >
      Fox Mulder
    </Link>
  </li>
  <li>
    FBI Agent
  </li>
  <li>
    Alive
  </li>
  <img
    src="https://vignette.wikia.nocookie.net/x-files/images/c/c0/TXF_Mulder.jpg/revision/latest/scale-to-width-down/310?cb=20171214040226"
  />
</ul>
`;
